peer.go

```diff
@@ -363,13 +371,22 @@ out:
 			break out
 		}
 
+		// Should wait for peers to compare their feature vectors
+		// and only then start message exchanges.
+		if !initialized {
+			if msg, ok := nextMsg.(*lnwire.Init); ok {
+				p.handleInitMsg(msg)
+				initialized = true
+			}
+			continue
```

Comment:
With this continue, it would be possible for nodes _not_ to send the `init` message as the first message. But it would also cause the _receiving_ to _miss_ the first message sent by the sender. The latter comment is a bug and should be fixed, but I think it's a little up to vote as what we'd like to do for the first option. 

A forwards compatible change (soft-fork) would be to allow not sending `init` as the first message in a sequence. Such a change would allow us to merge this into master or package up the fix in an upcoming `patch` release. 

On the other hand, a backwards incompatible change would be truer to the specification...